src/main.rs

```diff
@@ -73,37 +118,35 @@ fn server_routine(listener: &TcpListener, config: &mut Config, config_file: &Str
                 // do something with the message from the client
                 // and return a message
                 // is it a fatal error ?
-                let client_cmd = _parse_client_cmd(&formatted);
-                let client_arg = _parse_client_arg(&formatted);
-                // TODO
-                // let client_target_process: _parse_client_process(formatted);
-                match client_cmd {
-                    Ok(cmd) if cmd == "start" => {
-                        println!("start");
-                        println!("{:?}", client_arg);
-						config
-						.get_mut(&String::from("open_terminal"))
-						.unwrap()
-						.start(&String::from("open_terminal")); // error
-                    }
-                    Ok(cmd) if cmd == "stop" => {
-                        println!("stop");
-                        if client_arg.is_ok() {
-                            config
-                                .get_mut(&String::from("open_terminal"))
-                                .unwrap()
-                                .stop(&String::from("open_terminal")); // error
+                match parse_client_input(config, &formatted) {
+                    Ok((client_cmd, client_arg, client_process)) => {
+                        match client_cmd.as_str() {
+                            "start" => {
+                                config
+                                    .get_mut(&client_arg)
+                                    .unwrap()
+                                    .start(&client_arg, client_process);
```

Comment:
`let mut job = config.get_mut().unwrap()` can go before the match state to avoid code repetition

Reply:
How ? if I try to put it before the second match the syntax is not supported